go/src/lib.rs

```diff
@@ -46,17 +46,17 @@ pub type FailureCallback = unsafe extern "C" fn(
 ///
 /// It contains either a connection or an error. It is represented as a struct instead of a union for ease of use in the wrapper language.
 ///
-/// The struct is freed using `free_connection_response` to avoid memory leaks. This is achieved by having the external caller invoke this function.
+/// The struct is freed by the external caller by using `free_connection_response` to avoid memory leaks.
 #[repr(C)]
 pub struct ConnectionResponse {
     conn_ptr: *const c_void,
     connection_error_message: *const c_char,
 }
 
-/// The glide client.
+/// A `GlideClient` container.
```

Comment:
This is a nit, but I'm not a big fan of the "container" part.
size-and-complexity-wise, the runtime is a significantly bigger part of this struct than the client itself, so the client isn't the main contained thing here.
Logically,  its an adapter or a wrapper.